src/main/java/racingcar/domain/racing/RacingGameManagerImpl.java

```diff
@@ -0,0 +1,70 @@
+package racingcar.domain.racing;
+
+import camp.nextstep.edu.missionutils.Randoms;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.atomic.AtomicInteger;
+import racingcar.domain.car.Car;
+
+public class RacingGameManagerImpl implements RacingGameManager {
+
+  private static final int CRITICAL_VALUE = 4;
+
+
+  @Override
+  public void race(List<Car> cars) {
+    cars.forEach(car -> {
+      int random = generateRandomValue();
+      checkAndMoveForward(car, random);
+    });
+  }
+
+  @Override
+  public List<Car> getRaceResult(List<Car> cars) {
+    int maxPosition = getMaxPosition(cars);
+    List<Car> winningCar = processRaceResults(cars, maxPosition);
+    return winningCar;
+  }
+
+  private void checkAndMoveForward(Car car, int random) {
+    if (isForwardPossible(random)) {
+      car.plusPosition();
+    }
+  }
+
+  private int generateRandomValue() {
+    return Randoms.pickNumberInRange(0, 9);
+  }
```

Comment:
메서드로 만들 필요가 있나?? 의문인데 그리고 최소값, 최대값을 상수화 처리하기
```suggestion
  private int generateRandomValue() {
    return Randoms.pickNumberInRange(0, 9);
  }
```